### ISSUE TITLE
support for dictionary-type loss functions without explicitly declaring `None`

### ISSUE THREAD

**Issue by WJiang0123** [keras-team/keras#20243]
In previous versions of Keras, when providing a loss function as a dictionary for models with multiple outputs, the compileLoss function would automatically assign a default value for outputs without a defined loss. The relevant code was:
```            for name, yt, yp in zip(output_names, y_true, y_pred):
                if name in loss:
                    if loss[name]:
                        flat_losses.append(get_loss(loss[name], yt, yp))
                    else:
                        flat_losses.append(None)
                else:
                    flat_losses.append(None)
``` 
 However, in the latest version, flat_losses is now simply:
  ``` flat_losses = tree.flatten(loss) ``` 
Is there a way to reintroduce support for dictionary-type loss functions without explicitly declaring None for undefined losses? This change reduces flexibility in using dictionary-based loss definitions.

**Comment by fchollet** [comment]
Thanks, this is a reasonable ask. Would you be able to open a PR for the change?
